linc_convert/modalities/lsm/spool.py

```diff
@@ -0,0 +1,283 @@
+            all_shapes[rel_y, rel_z] = sx, sy, sz
+            expected_sx = sx
+            expected_sy[y_tile] = sy
+            expected_sz[z_tile] = sz
+            dtypes[dt].append((y_tile, z_tile))
+
+    # Ensure consistent dtype
+    if len(dtypes) != 1:
+        warnings.warn("Two or more dtypes in tiles:\n" + str(dict(dtypes)))
+    dtype = next(iter(dtypes))
+
+    # Ensure tiles's shapes are compatible
+    diff_sx = (all_shapes[:, :, 0] != expected_sx)
+    if diff_sx.any():
+        y_idxs, z_idxs = np.where(diff_sx)
+        raise ValueError(
+            f"Inconsistent x shapes at indices: {list(zip(y_idxs, z_idxs))}")
+    for y_tile in range(min_y_tile, max_y_tile + 1):
+        if y_tile not in expected_sy:
+            raise ValueError(f"Missing y tile {y_tile}")
+        diff_sy = (all_shapes[:, :, 1] != expected_sy[y_tile])
+        if diff_sy.any():
+            y_idxs, z_idxs = np.where(diff_sy)
+            raise ValueError(
+                f"Inconsistent y shapes at tiles: {list(zip(y_idxs, z_idxs))}")
+    for z_tile in range(min_z_tile, max_z_tile + 1):
+        if z_tile not in expected_sz:
+            raise ValueError(f"Missing z tile {z_tile}")
+        diff_sz = (all_shapes[:, :, 2] != expected_sz[z_tile])
+        if diff_sy.any():
+            y_idxs, z_idxs = np.where(diff_sz)
+            raise ValueError(
+                f"Inconsistent z shapes at tiles: {list(zip(y_idxs, z_idxs))}")
+
+    # Calculate full dataset dimensions
+    full_shape_x = expected_sx
+    full_shape_y = sum(expected_sy.values()) - (num_y_tiles - 1) * overlap
+    full_shape_z = sum(expected_sz.values())
+    fullshape = [full_shape_z, full_shape_y, full_shape_x]
+
+    # Initialize Zarr group and array
+    omz = zarr.storage.DirectoryStore(out)
+    omz = zarr.group(store=omz, overwrite=True)
+    print(out)
+    # Prepare chunking options
+    opt = {
+        "chunks": [chunk] * 3,
+        "dimension_separator": r"/",
+        "order": "F",
+        "dtype": np.dtype(dtype).str,
+        "fill_value": None,
+        "compressor": make_compressor(compressor, **compressor_opt),
+    }
+    # write first level
+    omz.create_dataset("0", shape=fullshape, **opt)
+    array = omz["0"]
+    print("Write level 0 with shape", fullshape)
+
+    # Populate Zarr array from tiles
+    for i, tile_info in enumerate(all_tiles_info):
+        rel_y, rel_z = tile_info.y - min_y_tile, tile_info.z - min_z_tile
+        dat = tile_info.reader.assemble_cropped()
+        if num_y_tiles != 1 and overlap != 0:
+            # if not first y tile, crop half overlapped rows at the beginning
+            if tile_info.y != min_y_tile:
+                dat = dat[:, overlap // 2:, :]
+            # if not last y tile, crop half overlapped rows at the end
+            # if overlap is odd, we need to crop an extra row
+            if tile_info.z != max_z_tile:
+                dat = dat[:, :-overlap // 2 - (overlap % 2), :]
```

Comment:
I might be misunderstanding the comments, but I think we need the following:

```suggestion
            # if not first y tile, crop half overlapped rows at the beginning
            if tile_info.y != min_y_tile:
                dat = dat[:, overlap // 2:, :]
            # if not last y tile, crop half overlapped rows at the end
            # if overlap is odd, we need to crop an extra column
            if tile_info.y != max_y_tile:
                dat = dat[:, :-overlap // 2 - (overlap % 2), :]
```